src/scheduler/gc_works.rs

```diff
@@ -131,12 +145,18 @@ impl<VM: VMBinding> GCWork<VM> for ReleaseMutator<VM> {
 }
 
 #[derive(Default)]
-pub struct ReleaseCollector;
+pub struct ReleaseCollector<W: CopyContext + WorkerLocal>(PhantomData<W>);
+
+impl<W: CopyContext + WorkerLocal> ReleaseCollector<W> {
+    pub fn new() -> Self {
+        ReleaseCollector(PhantomData)
+    }
+}
 
-impl<VM: VMBinding> GCWork<VM> for ReleaseCollector {
+impl<VM: VMBinding, W: CopyContext + WorkerLocal> GCWork<VM> for ReleaseCollector<W> {
     fn do_work(&mut self, worker: &mut GCWorker<VM>, _mmtk: &'static MMTK<VM>) {
         trace!("Release Collector");
-        worker.local().release();
+        unsafe { worker.local::<W>() }.release();
```

Comment:
We can simply use dynamic dispatch here and remove the `W: CopyContext + WorkerLocal` type argument in `ReleaseCollector`, `Release`, `PrepareCollector` and `Prepare`. The performance is probably not a problem. We only do the dynamic call twice for each worker, in a single gc.

Reply:
`worker.local()` is a `void*` pointer, and we cannot do dynamic dispatch on it. We rely on the `WorkerLocal` type the plan tells us when it creates this work packet, like here:
https://github.com/mmtk/mmtk-core/blob/0fc3a168a0f97bac9659e6fe856a5bdf8b3cc294/src/plan/semispace/global.rs#L92